hicexplorer/utilities.py

```diff
@@ -246,8 +246,9 @@ def expected_interactions(pSubmatrix):
     expected_interactions = np.zeros(pSubmatrix.shape[0])
     row, col = pSubmatrix.nonzero()
     distance = np.absolute(row - col)
-
-    occurences = np.zeros(pSubmatrix.shape[0])
+    #occurences = np.zeros(pSubmatrix.shape[0])
+    occurences = np.arange(1,pSubmatrix.shape[0]+1)
+    occurences = occurences[::-1] #devision is done taking all the memebers of a diagonal into account (as apposed to previous one where only non zero one were counted)
     for i, distance_ in enumerate(distance):
         expected_interactions[distance_] += pSubmatrix.data[i]
         occurences[distance_] += 1
```

Comment:
You don't need to count for the occurrences of non-zeros in this approach. The maximal number of possible contacts is given the length of the diagonal.